src/schema/schema-generator.ts

```diff
@@ -209,6 +217,15 @@ export abstract class SchemaGenerator {
               }
               return fields;
             },
+            resolveType: instance => {
+              const typeTarget = implementingObjectTypesTargets.find(
+                typeCls => instance instanceof typeCls,
+              );
+              if (!typeTarget) {
+                throw new InterfaceResolveTypeError(interfaceType);
+              }
+              return this.objectTypesInfo.find(type => type.target === typeTarget)!.type;
```

Comment:
I've never seen an exclamation mark used in syntax like this.

Reply:
It's the non-null assertion operator:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html